website/content/labs/lab4.md

```diff
@@ -106,35 +105,36 @@ Runtime errors make the program crash while it is running, and sometimes they ar
 
 `frame n` (where `n` is the frame number from the backtrace) allows you to go to the function in that frame and see what line in that particular function caused the error.
 
-Compile `task3.c` and run it. You will segfault. Now it’s time to debug! There are three errors in this file. For each error, answer the following questions in `task3.txt`. It is a good idea to fix one error before moving onto the next.
+Compile `task3.c` and run it. You will segfault. Now it’s time to debug! There are three errors in this file. For each error, answer the following questions in `task3.txt`. It is a good idea to fix one error before moving on to the next.
 
-1. Where was the segfault? Give the function, line number, and the line of code itself.
-2. What variables were involved?
-3. Why was this causing an error?
-4. What gdb commands did you use to debug this, and what did they help you with?
-5. Fix the errors in `task3.c` (do this in the C file directly)
+For each error, answer the following question on Gradescope:
 
+* Where was the segfault? Give the function, line number, and copy-paste the line of code itself. What was causing the segfault and how did you fix it?
+
+Make sure to also commit and push your fix for each segfault!
+
+You must also answer the following question for the entire task:
+
+* Describe the steps you took in gdb to narrow down the exact location of a segfault. You can also copy-paste the contents of one debugging session with gdb, with a brief note at the end explaining how that debugging session narrowed down the location of the error.
 
 ## Task 4: Logic Error Debugging
 
-(25 points)
+(20 points)
 
 Logic errors can happen even when your program compiles and runs, but may result in incorrect behavior. The commands we went through in the warmup can help you find logic errors by allowing actions such as stepping through the program line-by-line, printing variables, and seeing when variables are changed.
 
-Compile `task4.c`. As you can see from the print messages, the variables don’t have the expected values! Use gdb to find the five errors. For each error you find, answer the following questions in `task4.txt`.
+Compile `task4.c`. As you can see from the print messages, the variables don’t have the expected values! Use gdb to find the five errors. For each error you find, answer the following questions on Gradescope:
```

Comment:
Probably a bit clearer to change "Compile `task4.c`" to "Compile `task4.c` and run it"

Reply:
Fixed.